internal/jobmanager/jobmanager.go

```diff
@@ -0,0 +1,200 @@
+package jobmanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/edulinq/autograder/internal/lockmanager"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type JobOptions struct {
+	// Don't save anything.
+	DryRun bool `json:"dry-run"`
+
+	// Replace any entries currently in the cache,
+	// and do not return any cached entries (when not waiting for completion).
+	OverwriteCache bool `json:"overwrite-cache"`
```

Comment:
Given the more general nature of jobs, we may want to move away from the "cache" terminology,
and instead talk about "storing" and "retrieving" "records".